src/main/java/org/reaktivity/nukleus/http/internal/util/HttpUtil.java

```diff
@@ -101,7 +99,7 @@ public static boolean isPathValid(
         if (valid)
         {
             byte_loop:
-            for (int index = (originalCapacity < 8) ? 0 : capacity; capacity > 0; index++, capacity--)
+            for (int index = (originalCapacity < 8) ? 0 : originalCapacity - capacity; index < originalCapacity; index++)
```

Comment:
First, at the top, above the first `for` loop, remove `originalCapacity` and define `index` as follows:
```
int index = 0;
```
Then change first `for` loop to:
```
for (; capacity >= Long.BYTES; index += Long.BYTES, capacity -= Long.BYTES)
```
Now both `index` and `capacity` span both loops, with `capacity` being the remaining capacity and `index` being the index into the buffer, so the second `for` loop can become:
```
for (; capacity > 0; capacity--, index++)
```
The invariant throughout both loops is `index + capacity` always equals `path.capacity()`.